cmd/logger/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/kubeflow/kfserving/pkg/apis/serving/v1alpha2"
+	"net/http"
+	"net/url"
+	"os"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+
+	"github.com/kubeflow/kfserving/pkg/logger"
+	"github.com/pkg/errors"
+	"golang.org/x/net/http2"
+	"golang.org/x/net/http2/h2c"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var (
+	logUrl    = flag.String("log-url", "", "The URL to send request/response logs to")
+	port      = flag.String("port", "8080", "Logger port")
+	svcHost   = flag.String("svc-host", "0.0.0.0", "Service host")
+	svcPort   = flag.String("svc-port", "8081", "Service port")
+	workers   = flag.Int("workers", 5, "Number of workers")
+	sourceUri = flag.String("source-uri", "", "The source URI to use when publishing cloudevents")
+	logMode   = flag.String("log-mode", string(v1alpha2.LogAll), "Whether to log 'request', 'response' or 'all'")
+	modelId   = flag.String("model-id", "", "The model ID to add as header to log events")
+)
+
+func main() {
+	flag.Parse()
+
+	logf.SetLogger(logf.ZapLogger(false))
+	log := logf.Log.WithName("entrypoint")
+
+	if *logUrl == "" {
+		log.Info("log-url argument must not be empty.")
+		os.Exit(-1)
+	}
+
+	logUrlParsed, err := url.Parse(*logUrl)
+	if err != nil {
+		log.Info("Malformed log-url", "URL", *logUrl)
+		os.Exit(-1)
+	}
+	loggingMode := v1alpha2.LoggerMode(*logMode)
+	switch loggingMode {
+	case v1alpha2.LogAll, v1alpha2.LogRequest, v1alpha2.LogResponse:
+		break
+	default:
+		log.Info("Malformed log-mode", "type", *logMode)
+		os.Exit(-1)
+	}
+
+	if *sourceUri == "" {
+		*sourceUri = fmt.Sprintf("http://localhost:%s/", *port)
```

Comment:
Should we default this above in the args instead of this code?

Reply:
It depends on the port so don't think we can default.